types/nat-upnp/index.d.ts

```diff
@@ -0,0 +1,230 @@
+    UDN: string;
+    UPC: string;
+    serviceList?: { service: RawService | RawService[] };
+    deviceList?: { device: RawDevice | RawDevice[] };
+}
+
+export interface Device {
+    /**
+     * Get the available services on the network device
+     * @param types List of service types to lookf or
+     * @param callback
+     */
+    getService(
+        types: string[],
+        callback: CB<{
+            service: string;
+            SCPDURL: string;
+            controlURL: string;
+        }>,
+    ): void;
+    /**
+     * Parse out available services
+     * and devices from a root device
+     * @param info
+     * @returns the available devices and services in array form
+     */
+    parseDescription(info: {
+        device?: RawDevice;
+    }): {
+        services: RawService[];
+        devices: RawDevice[];
+    };
+    /**
+     * Perform a SSDP/UPNP request
+     * @param action the action to perform
+     * @param args arguments of said action
+     * @param callback Callback to be run when completed, or on error
+     */
+    run(action: string, args: string[], callback: CB<RawResponse>): void;
+}
+
+// Note for the SSDP class/interface
+// The implementation leads me to believe that
+// createSockets()
+// createSocket(interface)
+// parseResponse(response, addr)
+// are all private methods.
+// and have thusly not been typed in the interface
+
+export interface Ssdp extends EventEmitter {
+    /**
+     * Search for a SSDP compatible server on the network
+     * @param device Search Type (ST) header, specifying which device to search for
+     * @param promise An existing EventEmitter to emit event on
+     * @returns The event emitter provided in Promise, or a newly instantiated one.
+     */
+    search(device: string, promise?: EventEmitter): EventEmitter;
+    /**
+     * Close all sockets
+     */
+    close(): void;
+}
+
+//
+// Types for client.
+//
+export interface NewPortMappingOpts extends StandardOpts {
+    description?: string;
+    ttl?: number;
+}
+export type DeletePortMappingOpts = StandardOpts;
+export interface GetMappingOpts {
+    local?: boolean;
+    description?: RegExp | string;
+}
+export interface Mapping {
+    public: { host: string; port: number };
+    private: { host: string; port: number };
+    protocol: string;
+    enabled: boolean;
+    description: string;
+    ttl: number;
+    local: boolean;
+}
+
+/**
+ * Main client interface.
+ */
+export interface Client {
+    /**
+     * Create a new port mapping
+     * @param options Options for the new port mapping
+     * @param callback Callback to be run when completed, or on error
+     */
+    portMapping(options: NewPortMappingOpts, callback: CB<RawResponse>): void;
+    /**
+     * Remove a port mapping
+     * @param options Specify which port mapping to remove
+     * @param callback Callback to be run when completed, or on error
+     */
+    portUnmapping(options: DeletePortMappingOpts, callback: CB<RawResponse>): void;
```

Comment:
```suggestion
    portUnmapping(options: DeletePortMappingOpts, callback?: CB<RawResponse>): void;
```
https://github.com/indutny/node-nat-upnp/blob/9c80722df962ce1984946038e702e8eaccd0e97a/lib/nat-upnp/client.js#L56